### PR TITLE
Fix Rubocop rules

### PR DESCRIPTION
**Pull request by archanaserver**
https://github.com/theforeman/foreman_templates/pull/180

### CODE REVIEW

File: app/services/foreman_templates/export_result.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module ForemanTemplates
   class ExportResult
     attr_reader :template, :name, :template_file, :exported, :additional_info
 
-    def initialize(template, exported = true)
+    def initialize(template, exported: true)
```
Comment (ekohl) [inline app/services/foreman_templates/export_result.rb:5]: I think either this API change breaks things, or it's the one `parse_result`:
```
Failure:
test_0008_should force update locked template on import(Minitest::Result) [/home/jenkins/workspace/foreman_templates-pull-request/label/fast/ruby/2.7/plugin/test/functional/api/v2/template_controller_test.rb:110]:
Expected response to be a <2XX: success>, but was a <500: Internal Server Error>
Response body: {
  "error": {"message":"wrong number of arguments (given 1, expected 0)"}
}
```

But the commit `Fix Style/OptionalBooleanParameter cop` breaks things. It's not a safe cop where you can simply change things, just like I said in https://github.com/theforeman/foreman-installer/pull/883#discussion_r1331251537.

Comment (ofedoren) [reply app/services/foreman_templates/export_result.rb:5]: Even though boolean kw parameter looks nicer, I'd not change signatures just now to make rubocop happy.

File: app/services/foreman_templates/parse_result.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ def initialize(template_file)
       @template_file = template_file.split('/').last
     end
 
-    def to_h(verbose = false)
```
Comment (ofedoren) [inline app/services/foreman_templates/parse_result.rb:10]: Same here, let's stick with the old optional argument.

**Review by ofedoren — COMMENTED** [review]
Except for the boolean kw arguments rule, LGTM.

**Comment by ofedoren** [comment]
@archanaserver, could you please rebase?

File: app/services/foreman_templates/export_result.rb (hunk 1 above)
Comment (archanaserver) [reply app/services/foreman_templates/export_result.rb:5]: Do you want me to drop that changes related to `Style/OptionalBooleanParameter` cop?

File: .rubocop.yml [hunk 3]
```diff
@@ -98,6 +98,12 @@ Metrics:
...
   Enabled: false
 
+Lint/ConstantDefinitionInBlock:
+  Enabled: false
+
+Layout/ExtraSpacing:
+  Enabled: true
+
 # JobTemplates use this for inputs, we should probably whitelist classes
 Security/YAMLLoad:
```
[the remark this answers is not in the document]
Comment (archanaserver) [reply .rubocop.yml:108]: @ekohl what is this about? could you please brief more.

File: test/functional/api/v2/template_controller_test.rb [hunk 4]
```diff
@@ -26,7 +26,7 @@ class TemplateControllerTest < ::ActionController::TestCase
           FactoryBot.create(:provisioning_template, :name => 'export_test_template')
           post :export, params: { "repo" => tmpdir, "filter" => "^export_test_template", "negate" => true, "metadata_export_mode" => "keep" }
           assert_response :success
-          refute_includes(Dir.entries("#{tmpdir}/provisioning_templates/provision"), 'export_test_template.erb')
```
Comment (ofedoren) [inline test/functional/api/v2/template_controller_test.rb:29]: Hmm.. Is this necessary?

**Review by ofedoren — COMMENTED** [review]
Well, after rebasing it seems there is nothing left to fix. Maybe I'd treat this PR to ensure we resolved the issue Ewoud mentioned in https://github.com/theforeman/foreman_templates/pull/182/files#r1467571359

File: test/functional/api/v2/template_controller_test.rb (hunk 4 above)
Comment (archanaserver) [reply test/functional/api/v2/template_controller_test.rb:29]: I guess, I've skipped this while rebasing, lemme fix it!

**Pull request closed without merging**